packages/rainbowkit/src/components/ChainModal/ChainModal.tsx

```diff
@@ -24,48 +18,20 @@ export interface ChainModalProps {
 
 export function ChainModal({ onClose, open }: ChainModalProps) {
   const { chain: activeChain } = useNetwork();
-  const { chains, error: networkError, switchNetwork } = useSwitchNetwork();
+  const { chains, pendingChainId, reset, switchNetwork } = useSwitchNetwork({
+    onSettled: () => {
+      reset(); // reset mutation variables (eg. pendingChainId)
+      onClose();
+    },
+  });
+
   const { disconnect } = useDisconnect();
-  const { connector: activeConnector } = useAccount();
-  const [switchingToChain, setSwitchingToChain] = useState<number | null>();
   const titleId = 'rk_chain_modal_title';
   const mobile = isMobile();
   const rainbowkitChainsById = useRainbowKitChainsById();
   const unsupportedChain = activeChain?.unsupported ?? false;
   const chainIconSize = mobile ? '36' : '28';
 
-  const stopSwitching = useCallback(() => {
-    setSwitchingToChain(null);
-    onClose();
-  }, [onClose]);
-
-  useEffect(() => {
-    if (!activeConnector) {
-      return;
-    }
-
-    const stopSwitching = () => {
-      setSwitchingToChain(null);
-      onClose();
-    };
-
-    let provider: any;
-    activeConnector?.getProvider?.().then((provider_: any) => {
-      provider = provider_;
-      provider.on('chainChanged', stopSwitching);
-    });
-
-    return () => {
-      provider?.removeListener('chainChanged', stopSwitching);
-    };
-  }, [activeConnector, onClose, stopSwitching]);
-
-  useEffect(() => {
-    if (networkError && networkError.name === 'UserRejectedRequestError') {
-      stopSwitching();
-    }
-  }, [networkError, stopSwitching]);
-
```

Comment:
I'm pretty sure this is the same as this
```ts
const { chains, pendingChainId, reset, switchNetwork } = useSwitchNetwork({
    onSettled: () => {
      reset(); // reset mutation variables (eg. pendingChainId)
      onClose();
    },
  });
```

didn't find any differences in my testing, but easy to revert if the useEffects where proposital